Update to v22.1 (#30)
DevExpress-Examples/asp-net-core-richedit-customize-ribbon-ui-in-angular-app: README.md

[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T848130)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

> # Rich Text Editor for ASP.NET Core - How to customize the built-in ribbon in an Angular application

Reviewer: Again, should we capitalize Ribbon? I have a feeling (and a memory) that we should. The same also goes for the rest of this topic. Also, Ribbon what? the built-in Ribbon UI, more precisely.

For some reason, I'm unable to edit this line directly.

Author: ```suggestion
# Rich Text Editor for ASP.NET Core - How to customize the built-in Ribbon UI in an Angular application
```

> This example is a ready-to-use Angular client application that demonstrates how to customize the DevExpress [Rich Text Editor](https://docs.devexpress.com/AspNetCore/400373/office-inspired-controls/controls/rich-edit)'s ribbon. 

Reviewer: ```suggestion
This example is a ready-to-use Angular client application that demonstrates how to customize the DevExpress [Rich Text Editor](https://docs.devexpress.com/AspNetCore/400373/office-inspired-controls/controls/rich-edit)'s Ribbon UI.
```

![Customize Ribbon](customize-richedit-ribbon.png)

> The ribbon is a tabbed toolbar that displays command items on multiple tabs. Use the Rich Text Editor's [ribbon](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Options#js_devexpress_richedit_options_ribbon) option to access and customize the built-in ribbon. The [getTab](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Ribbon#js_devexpress_richedit_ribbon_gettab_id_) and [getItem](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.RibbonTab#js_devexpress_richedit_ribbontab_getitem_id_) methods allow you to access the ribbon's tabs and their items. Call the [insertTab](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Ribbon#js_devexpress_richedit_ribbon_inserttab_tab_) and [removeTab](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Ribbon#js_devexpress_richedit_ribbon_removetab_tab_) methods to modify the collection of tabs. To modify the item collection, call the [insertItem](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.RibbonTab#js_devexpress_richedit_ribbontab_insertitem_item_) and [removeItem](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.RibbonTab#js_devexpress_richedit_ribbontab_removeitem_item_) methods.

Reviewer: This paragraph has a series of issues:

1. Lacking capitalization of "Ribbon". I remember that we had a conversation where you agreed that we should capitalize it.
2. The first sentence does not outline the most important trait of the Ribbon UI - it groups commands according to their intended purpose.
3. Why do we use the mentioned `ribbon` option? To access and customize the built-in Ribbon UI. "Ribbon UI" is a more general term than "Ribbon" and includes all commands and UI elements.
4. the ribbon's tabs -- "Ribbon tabs" is way simpler and has the same effect.
5. The last sentence indirectly implies that we speak of items in a tab. Instead, we should speak of everything plainly and directly. As you may notice, I also outlined to what exactly the last mentioned methods belong.

**Bottom line** -- we should be as precise as possible in everything we do. If something can be understood incorrectly -- it surely will be. Here's my current take on this paragraph:

```suggestion
The Ribbon is a tabbed toolbar that groups command items on different tabs according to their purpose. Use the Rich Text Editor's [ribbon](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Options#js_devexpress_richedit_options_ribbon) option to access and customize the built-in Ribbon UI. The [getTab](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Ribbon#js_devexpress_richedit_ribbon_gettab_id_) and [getItem](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.RibbonTab#js_devexpress_richedit_ribbontab_getitem_id_) methods allow you to access Ribbon tabs and their items. Call the [insertTab](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Ribbon#js_devexpress_richedit_ribbon_inserttab_tab_) and [removeTab](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.Ribbon#js_devexpress_richedit_ribbon_removetab_tab_) methods to modify the collection of tabs. To modify the item collection in a tab, call its [insertItem](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.RibbonTab#js_devexpress_richedit_ribbontab_insertitem_item_) and [removeItem](https://docs.devexpress.com/AspNetCore/js-DevExpress.RichEdit.RibbonTab#js_devexpress_richedit_ribbontab_removeitem_item_) methods.
```

## Requirements

> * To use the Rich Text Editor control in an Angular application, you need to have a [Universal, DXperience, or ASP.NET subscription](https://www.devexpress.com/buy/net/).

Reviewer 2: ```suggestion
* To use the Rich Text Editor control in an Angular application, a [Universal, DXperience, or ASP.NET subscription](https://www.devexpress.com/buy/net/) is required.
```

> * Versions of the DevExpress npm packages should match.

Reviewer: I believe we should clarify this a bit. Whose versions should match? All DevExpress npm packages used in the current project, correct? If yes, we should be upfront with it.

Author: You are right, versions of all DevExpress npm packages used in the same project should match.

```suggestion
* Versions of all DevExpress npm packages used in this application should match.
```

Reviewer: OK.. but now the whole sentence looks a bit unnatural. What do you think of the following version?

```suggestion
* All Devexpress npm packages used in this application should have the same version.
```

> * All Devexpress npm packages used in this application should have the same version.

Reviewer 2: the same version of what? specify.

Author: Versions of all DevExpress npm packages should be the same. Be default, in this example the 22.1.9 version of the "devexpress-richedit" npm package is used. If you want to add another our package (for instance, the "devextreme" package) to this application, you must use the 22.1.9 version of this package.

Reviewer 2: ```suggestion
* All Devexpress npm packages used in this application should come from the same npm package version.
```

## Quick Start

Follow the steps below to run this application on your computer:

> 1. Open the *JS* folder in the command prompt and run the following command to install npm packages used in the application:

Reviewer: "Used in the application" implies definiteness for "npm packages". A definite article would have been too bold here, while "all" would have been a much better replacement:

```suggestion
1. Open the *JS* folder in the command prompt and run the following command to install all npm packages used in the application:
```
